src/sentry/models/exporteddata.py

```diff
@@ -57,18 +57,22 @@ def status(self):
         else:
             return ExportStatus.Valid
 
-    @property
-    def date_expired_string(self):
-        if self.date_expired is None:
-            return None
-        return self.date_expired.strftime("%-I:%M %p on %B %d, %Y (%Z)")
-
     @property
     def payload(self):
         payload = self.query_info.copy()
         payload["export_type"] = ExportQueryType.as_str(self.query_type)
         return payload
 
+    def get_date_string(self, date_field):
+        try:
+            if getattr(self, date_field) is None:
+                return None
+            else:
+                # Example: 12:21 PM on July 21, 2020 (UTC)
+                return getattr(self, date_field).strftime("%-I:%M %p on %B %d, %Y (%Z)")
+        except AttributeError:
+            pass
```

Comment:
@leeandher Might be worth logging an error here. Logging an error will also send it to Sentry. And I would return `None` as well.

Reply:
Refactored it to a `@classmethod`, so it shouldn't need a try/catch or anything.